cuqi/experimental/mcmc/_langevin_algorithm.py

```diff
@@ -61,15 +62,15 @@ class ULANew(SamplerNew): # Refactor to Proposal-based sampler?
     # TODO: update demo once sampler merged
     """
 
-    _STATE_KEYS = SamplerNew._STATE_KEYS.union({'current_target_logd', 'scale', 'current_target_grad'})
+    _STATE_KEYS = SamplerNew._STATE_KEYS.union({'scale', 'current_target_grad'})
 
     def __init__(self, target, scale=1.0, **kwargs):
 
         super().__init__(target, **kwargs)
 
         self.scale = scale
         self.current_point = self.initial_point
-        self.current_target_logd = self.target.logd(self.current_point)
+        self.current_target_logd = self._eval_target_logd(self.current_point)
```

Comment:
Good point! thanks!
We simplified this logic now so that ULANew does not store the state variable current_target_logd and this only needed a few line changes.